src/Components/Web/test/Forms/InputRadioTest.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.AspNetCore.Components.Test.Helpers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    public class InputRadioTest
+    {
+        [Theory]
+        [InlineData(" ")]
+        [InlineData(null)]
+        public async Task ThrowsOnFirstRenderIfInvalidNameSuppliedWithoutGroup(string name)
+        {
+            var model = new TestModel();
+            var rootComponent = new TestInputRadioHostComponent<TestEnum>
+            {
+                EditContext = new EditContext(model),
+                InnerContent = RadioButtonsWithoutGroup(name, () => model.TestEnum)
+            };
+
+            var testRenderer = new TestRenderer();
+            var componentId = testRenderer.AssignRootComponentId(rootComponent);
+
+            var ex = await Assert.ThrowsAsync<InvalidOperationException>(() => testRenderer.RenderRootComponentAsync(componentId));
+            Assert.Contains($"requires either an explicit 'name' attribute", ex.Message);
+        }
+
+        [Theory]
+        [InlineData(" ")]
+        [InlineData(null)]
+        public async Task GeneratesNameGuidWhenInvalidNameSuppliedWithGroup(string name)
+        {
+            var model = new TestModel();
+            var rootComponent = new TestInputRadioHostComponent<TestEnum>
+            {
+                EditContext = new EditContext(model),
+                InnerContent = RadioButtonsWithGroup(name, () => model.TestEnum)
+            };
+
+            var testRenderer = new TestRenderer();
+            var componentId = testRenderer.AssignRootComponentId(rootComponent);
+            await testRenderer.RenderRootComponentAsync(componentId);
+            var inputRadioComponents = FindInputRadioComponents(testRenderer.Batches.Single());
+
+            Assert.All(inputRadioComponents, inputRadio => Assert.True(Guid.TryParseExact(inputRadio.GroupName, "N", out _)));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAllInputRadioGenerators))]
+        public async Task NameAttributeExistsWhenValidNameSupplied(InputRadioGenerator generator)
```

Comment:
Could we turn these in to separate tests that state what scenario is being tested?